File: TestDomain/AnomalyPhoto.cs
using lvl.Ontology;
using lvl.Ontology.Authorization;
using System.ComponentModel.DataAnnotations;

namespace lvl.TestDomain
{
    [Role(Actions.Read, "nasa")]
    [OwnedByUser(nameof(TakenByUserId), Actions.Read)]
    public class AnomalyPhoto : Entity, IAggregateRoot
    {
        [Required]
        public string Name { get; set; }

        public int Views { get; set; }

        [Required]
        public string TakenByUserId { get; set; }
    }
}

File: front-end/demo-app/models/anomaly-photo.ts
import { Entity, IAggregateRoot, Required } from '../../src';

export class AnomalyPhoto extends Entity implements IAggregateRoot {
    @Required() public name: string;
    public views: number;
    @Required() public takenByUserId: string;

    constructor(options?: IAnomalyPhotoOptions) {
        options = options || {};
        super(options);
        this.name = options!.name;
        this.views = options!.views;
        this.takenByUserId = options!.takenByUserId;
    }
}

interface IAnomalyPhotoOptions {
    name?: string;
    views?: number;
    takenByUserId?: string;
    id?: number;
}
